Jenkinsfile

```diff
@@ -45,11 +58,12 @@ def rocmtestnode(Map conf) {
             cmake -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DBUILD_DEV=On -DCMAKE_EXECUTE_PROCESS_COMMAND_ECHO=STDOUT -DMIGRAPHX_DISABLE_VIRTUAL_ENV=ON ${flags} ..
             git diff
             git diff-index --quiet HEAD || (echo "Git repo is not clean after running cmake." && exit 1)
-            make -j\$(nproc) generate VERBOSE=1
+            #make -j\$(nproc) generate VERBOSE=1
             git diff
             git diff-index --quiet HEAD || (echo "Generated files are different. Please run make generate and commit the changes." && exit 1)
-            make -j\$(nproc) all package check VERBOSE=1
+            #make -j\$(nproc) all package check VERBOSE=1
             md5sum ./*.deb
+            echo ${GIT_COMMIT_SHA}  > "${name}_PR"
```

Comment:
I dont think storing it locally in the workspace is useful since it wont exist on every node. Instead you should check the github commit status, something like this:

```groovy
    withCredentials([string(credentialsId: "${env.migraphx_ci_creds}", variable: 'GITHUB_TOKEN')]) {
        def commit_hash = sh(script: 'git rev-parse --short HEAD', returnStdout: true).trim()
        def response = sh(
            script: """curl -H "Accept: application/vnd.github+json" \
                           -H "Authorization: token ${GITHUB_TOKEN}" \
                           https://api.github.com/repos/ROCm/AMDMIGraphX/commits/${commit_hash}/status""",
            returnStdout: true
        ).trim()
        def statuses = response.statuses
        def contextStatus = statuses.find { it.context == "Jenkins - ${variant}"
        env.COMMIT_PASSED = contextStatus != null && contextStatus.state == 'success'
    }
```

Reply:
interesting, I was trying to use archieveAritfact since it would be shared across nodes, but I hadn't figured out a way to pull the file down.  I like your idea and will try that